### ISSUE TITLE
Local storage in ios gone after app update

### ISSUE THREAD

**Issue by cis2131** [thunkable/thunkable-issues#637]
After releasing a new update to my app, the local storage that was in users app before the update seems to be deleted.

The project has been duplicated a couple of times, but I have kept working on the main project.

What can be the reason for this, and how can I make sure to send out an update that will preserve local storage in the future?.

**Comment by AmigoDeluxe** [comment]
I experienced the same. Sucks big time. I hope it is the last time it happens.

**Comment by towball** [comment]
Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update

**Comment by cis2131** [comment]
> Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update

No. I changed my app to use firebase instead, as i could not take the chance that users would lose their data once more.

**Comment by towball** [comment]
> > Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update
> 
> No. I changed my app to use firebase instead, as i could not take the chance that users would lose their data once more.

Do yo use thunkable or something else? The only problem with firebase is users need an internet connection or can it be set up so if the phone does not have an internet connection it still shows the last of the users entry. And if changed it updates firebase when an internet connection returns

**Comment by cis2131** [comment]
> > > Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update
> > 
> > 
> > No. I changed my app to use firebase instead, as i could not take the chance that users would lose their data once more.
> 
> Do yo use thunkable or something else? The only problem with firebase is users need an internet connection or can it be set up so if the phone does not have an internet connection it still shows the last of the users entry. And if changed it updates firebase when an internet connection returns

Yes. I use thunkable, so i had to tell users that internet connection was required.
You can always load data from firebase, and store it in local DB, or local Storage, and then check if device is online and if not load the data from local, then set a variable so the app knows which copy of the data is the newest. But you will have to build this function yourselves.

**Comment by towball** [comment]
> > > > Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update
> > > 
> > > 
> > > No. I changed my app to use firebase instead, as i could not take the chance that users would lose their data once more.
> > 
> > 
> > Do yo use thunkable or something else? The only problem with firebase is users need an internet connection or can it be set up so if the phone does not have an internet connection it still shows the last of the users entry. And if changed it updates firebase when an internet connection returns
> 
> Yes. I use thunkable, so i had to tell users that internet connection was required.
> You can always load data from firebase, and store it in local DB, or local Storage, and then check if device is online and if not load the data from local, then set a variable so the app knows which copy of the data is the newest. But you will have to build this function yourselves.

Thanks. Still have the issue of updating and changing to firebase and ensuring users can still retrieve data. I have logged a case with Google Play and currently chatting to Thunkable to resolve.
So your App was updating and not retrieving local storage. And when you roll back it retrieves local storage????

**Comment by cis2131** [comment]
> > > > > Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update
> > > > 
> > > > 
> > > > No. I changed my app to use firebase instead, as i could not take the chance that users would lose their data once more.
> > > 
> > > 
> > > Do yo use thunkable or something else? The only problem with firebase is users need an internet connection or can it be set up so if the phone does not have an internet connection it still shows the last of the users entry. And if changed it updates firebase when an internet connection returns
> > 
> > 
> > Yes. I use thunkable, so i had to tell users that internet connection was required.
> > You can always load data from firebase, and store it in local DB, or local Storage, and then check if device is online and if not load the data from local, then set a variable so the app knows which copy of the data is the newest. But you will have to build this function yourselves.
> 
> Thanks. Still have the issue of updating and changing to firebase and ensuring users can still retrieve data. I have logged a case with Google Play and currently chatting to Thunkable to resolve.
> So your App was updating and not retrieving local storage. And when you roll back it retrieves local storage????

No all my users lost their data, and had to enter it again. But I have made a backup function in the app, so if they had used that function then they could just restore their data.

**Comment by towball** [comment]
> > > > > > Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update
> > > > > 
> > > > > 
> > > > > No. I changed my app to use firebase instead, as i could not take the chance that users would lose their data once more.
> > > > 
> > > > 
> > > > Do yo use thunkable or something else? The only problem with firebase is users need an internet connection or can it be set up so if the phone does not have an internet connection it still shows the last of the users entry. And if changed it updates firebase when an internet connection returns
> > > 
> > > 
> > > Yes. I use thunkable, so i had to tell users that internet connection was required.
> > > You can always load data from firebase, and store it in local DB, or local Storage, and then check if device is online and if not load the data from local, then set a variable so the app knows which copy of the data is the newest. But you will have to build this function yourselves.
> > 
> > 
> > Thanks. Still have the issue of updating and changing to firebase and ensuring users can still retrieve data. I have logged a case with Google Play and currently chatting to Thunkable to resolve.
> > So your App was updating and not retrieving local storage. And when you roll back it retrieves local storage????
> 
> No all my users lost their data, and had to enter it again. But I have made a backup function in the app, so if they had used that function then they could just restore their data.

Hey do u have example blocks for backup routine?

**Comment by cis2131** [comment]
> > > > > > > Did you resolve this, I am using Thunkable to create apps and when I load a new version to the stores it updates and local storage data in the App is not showing after loading the update
> > > > > > 
> > > > > > 
> > > > > > No. I changed my app to use firebase instead, as i could not take the chance that users would lose their data once more.
> > > > > 
> > > > > 
> > > > > Do yo use thunkable or something else? The only problem with firebase is users need an internet connection or can it be set up so if the phone does not have an internet connection it still shows the last of the users entry. And if changed it updates firebase when an internet connection returns
> > > > 
> > > > 
> > > > Yes. I use thunkable, so i had to tell users that internet connection was required.
> > > > You can always load data from firebase, and store it in local DB, or local Storage, and then check if device is online and if not load the data from local, then set a variable so the app knows which copy of the data is the newest. But you will have to build this function yourselves.
> > > 
> > > 
> > > Thanks. Still have the issue of updating and changing to firebase and ensuring users can still retrieve data. I have logged a case with Google Play and currently chatting to Thunkable to resolve.
> > > So your App was updating and not retrieving local storage. And when you roll back it retrieves local storage????
> > 
> > 
> > No all my users lost their data, and had to enter it again. But I have made a backup function in the app, so if they had used that function then they could just restore their data.
> 
> Hey do u have example blocks for backup routine?

Sure. but think this thread is not for this kind of discussion. send me a PM at  Claus_Holbech1 in thunkable community. Then i can send you how i made that.

**Comment by towball** [comment]
> I experienced the same. Sucks big time. I hope it is the last time it happens.

Did you find a solution?

**Comment by towball** [comment]
@jane-d-alt has there been any progress on this. I am about to do a full rewrite that will not update the previous live version. Better a new version than upset current users
